preciceconfigchecker/rule.py

```diff
@@ -1,27 +1,55 @@
-from typing import List, Callable
-from violation import Violation, Problem
+from abc import ABC, abstractmethod
+from typing import List
 from severity import Severity
 
 
-class Rule:
-    def __init__(self, problem:Problem, severity:Severity = Severity.INFO) -> None:
-        self.problem = problem
+class Rule(ABC):
+    numbers:int = 0 #static attribute: do not use 'self.numbers', use 'Rule.numbers'
+
+    def __init__(self, severity:Severity, problem:str) -> None:
         self.severity = severity
-        self.violations:List[Violation] = []
+        self.problem = problem
+        self.rule:str
+        self.violations:List[str] = []
+        self.result:List[tuple] = []
 
-    def check_with(self, check_method:Callable, args:tuple) -> None:
-        result:List = check_method(*args)
-        for data in result:
-            violation = Violation(self.problem, data)
-            self.violations.append(violation)
+    @abstractmethod
+    def check_method(self) -> List[tuple]:
```

Comment:
This checks if the rule was followed, correct? If so, maybe name it check_rule; check_method sounds like some method is getting checked

Reply:
check out the second conversation above between me and s3n-w6i.

Reply:
That does not mention my issue, I think the method should be renamed

Reply:
How about check_for_violatons?

Reply:
done